gist.go

```diff
@@ -0,0 +1,86 @@
+package model
+
+// Typed constants
+type ChannelType string
+
+// CamelCase Constant Names
+const (
+	ChannelTypeOpen    ChannelType = "O"
+	ChannelTypePrivate ChannelType = "P"
+)
+
+type Channel struct {
+	// Member variables are not exported to ensure immutability.
+	id       string `json:"id"`
+	name     string `json:"name"`
+	createAt int64  `json:"create_at"`
+	// etc...
+
+	// Question: what do we do when there's a semantically meaningful
+	// "null" value for a member variable?
+	schemeID *string `json:"scheme_id"`
+
+	// Question: is this the best way of storing props and similar map/list data?
+	props map[string]interface{} `json:"props" db:"-"`
+}
+
+// Getters to access the member variables are provided. These *never* return pointers.
+// Correct capitalisation of initialisms
+func (channel *Channel) GetID() string {
```

Comment:
Should we remove `Get` prefix?

```suggestion
func (channel *Channel) ID() string {
```

Reply:
Yeah, I think it's generally considered to be more idiomatic in Go to not have  the `Get`, right?

Reply:
Correct. https://golang.org/doc/effective_go.html#Getters

from the link above
```
owner := obj.Owner()
if owner != user {
    obj.SetOwner(user)
}
```

Reply:
Yes, [net/http#Request](https://pkg.go.dev/net/http?tab=doc#Request) is a good example from Go stdlib.